rialto_airflow/publish/openaccess.py

```diff
@@ -0,0 +1,159 @@
+import logging
+from csv import DictWriter
+from pathlib import Path
+
+from sqlalchemy import select, func
+
+from rialto_airflow.database import get_session, Publication, Author, Funder
+
+
+def write_publications(snapshot) -> Path:
+    """
+    Write a CSV of publications including their funding information.
+    """
+    col_names = [
+        "doi",
+        "pub_year",
+        "apc",
+        "open_access",
+        "types",
+        "funders",
+        "federally_funded",
+    ]
+
+    csv_path = snapshot.path / "publications.csv"
+    logging.info(f"started writing publications {csv_path}")
+
+    with csv_path.open("w") as output:
+        csv_output = DictWriter(output, fieldnames=col_names)
+        csv_output.writeheader()
+
```

Comment:
Perhaps it would be helpful to have a comment explaining what this query is doing? I wasn't sure how readable it was.

Reply:
I'm trying to understand how `func.jsonb_agg_strict(Funder.name).label("funders"),` works and with the `group_by` using type, so if there's anything to explain there, that could be helpful.

Reply:
I added some comments to both blocks of SQL. Do they help at all?

Reply:
Yes, that helps, thanks!